flask_server.py

```diff
@@ -73,33 +73,40 @@ def index():
 
 @app.route("/settings/", methods=['GET','POST'])
 def settings():
-    login_error = False
     settings = get_settings()    # Gets the settings
+    if not "session_id" in settings: return redirect("/login")
     folder_path = settings["folder_path"] if "folder_path" in settings else "ig_media"
-    extra_ids = settings["extra_ids"] if "extra_ids" in settings else []
+    extra_ids = settings["extra_ids"] if "extra_ids" in settings else [] 
     if request.method == "POST":
         updated_settings = settings
         for setting in request.form:
-            if len(request.form[setting]) > 0 and not ("username" or "password") in request.form:
+            if len(request.form[setting]) > 0:
                 if setting == "extra_ids":
                     updated_settings[setting] = request.form[setting].splitlines()
                     continue
                 updated_settings[setting] = request.form[setting]    
         save_settings(updated_settings)
         extra_ids = updated_settings["extra_ids"]
-        if ("username" and "password") in request.form:
-            login_error = store_session_id(request.form["username"], request.form["password"])
-    logged_in = check_login_status()
-    return render_template("settings.html", folder_path = folder_path, extra_ids = extra_ids, disclaimer = {"logged_in": logged_in, "login_error": login_error})
+    return render_template("settings.html", folder_path = folder_path, extra_ids = extra_ids)
 
+
+@app.route("/login", methods=['GET','POST'])
+def login():
+    login_error = False
+    if request.method == "POST":
+        print("POSTTT")
+        if ("username" and "password") in request.form:
+                login_error = store_session_id(request.form["username"], request.form["password"])
+                print(login_error)
+                if login_error == False: return redirect("/settings")
```

Comment:
Rename the variable above to 
login_successfull and do
if login_successfull: return ....